README.md

```diff
@@ -1,2 +1,102 @@
 # node-problem-detector
-This is a place for various problem detectors running on the Kubernetes nodes.
+node-problem-detector aims to make various node problems visible to Kubernetes 
+control-plane. It is a [DaemonSet](http://kubernetes.io/docs/admin/daemons/)
+detecting node problems and reporting them to apiserver. Now it is running as
+a [Kubernetes Addon](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons)
+enabled by default in the GCE cluster.
+
+# Background
+There are tons of node problems could possibly affect the pods running on the
+node such as:
+* Hardware Issues: Bad cpu, memory or disk;
+* Kernel Issues: Kernel deadlock, corrupted file system;
+* Container Runtime Issues: Unresponsive runtime daemon;
+* ...
+
+Currently these problems are invisible to the control plane, so Kubernetes will
+continue scheduling pods to the bad nodes.
+
+To solve this problem, we introduced this new daemon **node-problem-detector** to
+detect node problems and make them visible to the control-plane. Once control-plane
```

Comment:
This is a little bit misleading. The user might think a single node-problem-detector will detect all potential node problems, which is not the goal here. NodeProblemDetector is collecting the node problems from various daemons (only KernelMonitor for now, and rest of daemons are TBD).